ark/segmentation/marker_quantification_test.py

```diff
@@ -549,47 +550,58 @@ def test_generate_cell_data_tree_loading():
-            label_data=cell_masks,
-            compartment_names=['whole_cell']
-        )
+        for fov in range(cell_masks.shape[0]):
+            fov_whole_cell = cell_masks[fov, :, :, 0]
+            fov_nuclear = cell_masks[fov, :, :, 1]
+            io.imsave(os.path.join(temp_dir, 'fov%d_feature_0.tif' % fov), fov_whole_cell)
+            io.imsave(os.path.join(temp_dir, 'fov%d_feature_1.tif' % fov), fov_nuclear)
 
-        with pytest.raises(ValueError):
+        with pytest.raises(FileNotFoundError):
             # specifying fovs not in the original segmentation mask
             marker_quantification.generate_cell_table(
-                segmentation_labels=segmentation_masks.loc[["fov1"]], tiff_dir=tiff_dir,
-                img_sub_folder=img_sub_folder, is_mibitiff=False, fovs=["fov1", "fov2"],
+                segmentation_dir=temp_dir, tiff_dir=tiff_dir,
+                img_sub_folder=img_sub_folder, is_mibitiff=False, fovs=["fov2", "fov3"],
                 batch_size=5)
 
         # generate sample norm and arcsinh data for all fovs
         norm_data, arcsinh_data = marker_quantification.generate_cell_table(
-            segmentation_labels=segmentation_masks, tiff_dir=tiff_dir,
+            segmentation_dir=temp_dir, tiff_dir=tiff_dir,
             img_sub_folder=img_sub_folder, is_mibitiff=False, fovs=None, batch_size=2)
 
         assert norm_data.shape[0] > 0 and norm_data.shape[1] > 0
         assert arcsinh_data.shape[0] > 0 and arcsinh_data.shape[1] > 0
 
         # generate sample norm and arcsinh data for a subset of fovs
         norm_data, arcsinh_data = marker_quantification.generate_cell_table(
-            segmentation_labels=segmentation_masks, tiff_dir=tiff_dir,
+            segmentation_dir=temp_dir, tiff_dir=tiff_dir,
             img_sub_folder=img_sub_folder, is_mibitiff=False, fovs=fovs_subset, batch_size=2)
 
         assert norm_data.shape[0] > 0 and norm_data.shape[1] > 0
         assert arcsinh_data.shape[0] > 0 and arcsinh_data.shape[1] > 0
 
         # generate sample norm and arcsinh data for a subset of fovs
         norm_data, arcsinh_data = marker_quantification.generate_cell_table(
-            segmentation_labels=segmentation_masks, tiff_dir=tiff_dir,
+            segmentation_dir=temp_dir, tiff_dir=tiff_dir,
             img_sub_folder=img_sub_folder, is_mibitiff=False, fovs=fovs_subset_ext, batch_size=2)
 
         assert norm_data.shape[0] > 0 and norm_data.shape[1] > 0
         assert arcsinh_data.shape[0] > 0 and arcsinh_data.shape[1] > 0
 
+        # test nuclear_counts True
+        norm_data, arcsinh_data = marker_quantification.generate_cell_table(
+            segmentation_dir=temp_dir, tiff_dir=tiff_dir,
+            img_sub_folder=img_sub_folder, is_mibitiff=False, fovs=fovs_subset,
+            batch_size=2, nuclear_counts=True)
+
+        assert norm_data.shape[0] > 0 and norm_data.shape[1] > 0
+        assert arcsinh_data.shape[0] > 0 and arcsinh_data.shape[1] > 0
```

Comment:
This is just a copy paste of the same tests from without nuclear counts. This doesn't check that nuclear counts are returned. We should check that the shape is correct given that nuclear counts were generated.